src/Lean/Elab/BuiltinCommand.lean

```diff
@@ -138,6 +203,117 @@ private partial def elabChoiceAux (cmds : Array Syntax) (i : Nat) : CommandElabM
     aliases := aliases.push (currNamespace ++ id, declName)
   modify fun s => { s with env := aliases.foldl (init := s.env) fun env p => addAlias env p.1 p.2 }
 
+/-- Makes names from other namespaces visible without writing the namespace prefix.
+
+Names that are made available with `open` are visible within the current `section` or `namespace`
+block. The `open` command can be used in few different ways:
+
+* `open Some.Namespace.Path1 Some.Namespace.Path2` makes all names in `Some.Namespace.Path1` and
+  `Some.Namespace.Path2` available without the prefix, so that `Some.Namespace.Path1.x` and
+  `Some.Namespace.Path2.y` can be referred to by writing only `x` and `y`.
+
+  This command only makes *public* names visible, *i.e.* it does not bring `protected` names in
+  scope.
+
+* `open Some.Namespace.Path (def1 def2)` only makes `Some.Namespace.Path.def1` and
+  `Some.Namespace.Path.def2` available without the full prefix, so `Some.Namespace.Path.def3` would
+  be unaffected.
+
+  This works even if `def1` and `def2` are `protected`.
+
+* `open Some.Namespace.Path renaming def1 → def1', def2 → def2'` same as `open Some.Namespace.Path
+  (def1 def2)` but `def1`/`def2`'s names are changed to `def1'`/`def2'`.
+
+  This works even if `def1` and `def2` are `protected`.
+
+* `open Some.Namespace.Path hiding def1 def2` opens everything in `Some.Namespace.Path` except
+  `def1` and `def2`.
+
+  Only makes public names visible.
+
+* `open scoped Some.Namespace.Path1 Some.Namespace.Path2` opens [scoped instances], notations, and
```

Comment:
Do all of the commands above also do this?

Reply:
So I moved the `open ... hiding ...` right after the `open Namespace` as the former is a refinement of the latter. I also mention upfront that it only works on non-protected names, as in one of your remarks on `open Namespace`.

```markdown
* `open Some.Namespace.Path1 Some.Namespace.Path2` makes all non-protected names in
  `Some.Namespace.Path1` and `Some.Namespace.Path2` available without the prefix, so that
  `Some.Namespace.Path1.x` and `Some.Namespace.Path2.y` can be referred to by writing only `x` and
  `y`.

* `open Some.Namespace.Path hiding def1 def2` opens all non-protected names in `Some.Namespace.Path`
  except `def1` and `def2`.
```

Reply:
To be clear, this comment is attached to the line about `open scoped`.

Reply:
My bad. You're right, `open Namespace` also brings in scoped instances and notations

Reply:
I pushed a fix for this. Before the variants, we now clarify that

```markdown
Names that are made available with `open` are visible within the current `section` or `namespace`
block. This makes referring to (type) definitions and theorems easier, but note that it can also make
[scoped instances], notations, and attributes from a different namespace available.
```